src/actions/actions.ts

```diff
@@ -402,6 +402,49 @@ class CommandRegister extends BaseCommand {
 }
 
 @RegisterAction
+class CommandInsertRegisterContent extends BaseCommand {
+  modes = [ModeName.Insert];
+  keys = ["ctrl+r", "<character>"];
+  isCompleteAction = false;
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    vimState.recordedState.registerName = this.keysPressed[1];
+    const register = await Register.get(vimState);
+    let text: string;
+
+    if (typeof register.text === "object") {
+       text = (register.text as string []).join("\n");
```

Comment:
The cast is actually unnecessary here due to the magic of typescript 2.0 and discriminated union types! :)